src/components/Button/Button.tsx

```diff
@@ -91,9 +92,13 @@ const Button: React.FC<ButtonProps> = ({
       style={[{height: ButtonTheme.height}, StyleSheet.flatten(style)]}
       {...props}>
       {!!leftIcon && leftIcon}
-      <Label {...textProps} style={textStyle} color={textColor ?? ButtonTheme.textColor}>
-        {text}
-      </Label>
+      {children ? (
```

Comment:
I think you should check typeof children. if children is string:
it will return ``<Label>{children}</Label>``
else:
return ``{children}``
and remove text props